ports/stm32f4/common-hal/busio/UART.c

```diff
@@ -207,24 +205,32 @@ void common_hal_busio_uart_construct(busio_uart_obj_t *self,
     if(USARTx==USART1) { 
         reserved_uart[0] = true;
         __HAL_RCC_USART1_CLK_ENABLE();
+        HAL_NVIC_SetPriority(USART1_IRQn, 0, 1);
+        HAL_NVIC_EnableIRQ(USART1_IRQn);
     } 
     #endif
     #ifdef UART2
     if(USARTx==USART2) { 
         reserved_uart[1] = true;
         __HAL_RCC_USART2_CLK_ENABLE();
+        HAL_NVIC_SetPriority(USART2_IRQn, 0, 1);
+        HAL_NVIC_EnableIRQ(USART2_IRQn);
     } 
     #endif
     #ifdef USART3
     if(USARTx==USART3) { 
         reserved_uart[2] = true;
         __HAL_RCC_USART3_CLK_ENABLE();
+        HAL_NVIC_SetPriority(USART3_IRQn, 0, 1);
+        HAL_NVIC_EnableIRQ(USART3_IRQn);
     } 
     #endif
     #ifdef UART4
     if(USARTx==UART4) { 
         reserved_uart[3] = true;
         __HAL_RCC_UART4_CLK_ENABLE();
+        HAL_NVIC_SetPriority(UART4_IRQn, 0, 1);
+        HAL_NVIC_EnableIRQ(UART4_IRQn);
     } 
     #endif
     #ifdef UART5
```

Comment:
I'm not sure there is a way to condense this. To make this more readable please factor it out into a helper function.